mtags/src/main/scala-2/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -324,82 +395,110 @@ trait MatchCaseCompletions { this: MetalsGlobal =>
-  ): TextEditMember = {
-    sym.info // complete
-    val isModuleLike = fsym.hasModuleFlag || fsym.hasJavaEnumFlag
-    if (sym.isCase || isModuleLike) {
-      // Syntax for deconstructing the symbol as an infix operator, for example `case head :: tail =>`
-      val isInfixEligible =
-        context.symbolIsInScope(sym) ||
-          autoImports.nonEmpty
-      val infixPattern: Option[String] =
-        if (
-          isInfixEligible &&
-          sym.isCase &&
-          !Character.isUnicodeIdentifierStart(sym.decodedName.head)
-        ) {
-          sym.primaryConstructor.paramss match {
-            case (a :: b :: Nil) :: _ =>
-              Some(
-                s"${a.decodedName} ${sym.decodedName} ${b.decodedName}"
-              )
-            case _ => scala.None
-          }
-        } else {
-          scala.None
-        }
-      val pattern = infixPattern.getOrElse {
-        // Fallback to "apply syntax", example `case ::(head, tail) =>`
-        val suffix =
-          if (isModuleLike) ""
-          else {
-            sym.primaryConstructor.paramss match {
-              case Nil => "()"
-              case head :: _ =>
-                head
-                  .map(param => Identifier(param.name))
-                  .mkString("(", ", ", ")")
-            }
-          }
-        name + suffix
-      }
-      val label = s"case $pattern =>"
+      context: Context,
+      clientSupportsSnippets: Boolean,
+      patternOnly: Option[String] = None,
+      hasBind: Boolean = false
+  ) {
+
+    def toCaseMember(
+        name: String,
+        sym: Symbol,
+        fsym: Symbol,
+        autoImports: List[l.TextEdit]
+    ): TextEditMember = {
+      sym.info
+      val isModuleLike = fsym.hasModuleFlag || fsym.hasJavaEnumFlag
+      val pattern =
+        if ((sym.isCase || isModuleLike) && !hasBind) {
+          val isInfixEligible =
+            context.symbolIsInScope(sym) ||
+              autoImports.nonEmpty
+          if (
+            isInfixEligible &&
+            sym.isCase &&
+            !Character.isUnicodeIdentifierStart(sym.decodedName.head)
+          )
+            // Deconstructing the symbol as an infix operator, for example `case head :: tail =>`
+            tryInfixPattern(sym).getOrElse(
+              unapplyPattern(sym, name, isModuleLike)
+            )
+          else
+            unapplyPattern(
+              sym,
+              name,
+              isModuleLike
+            ) // Apply syntax, example `case ::(head, tail) =>`
+        } else
+          typePattern(
+            sym,
+            name
+          ) // Symbol is not a case class with unapply deconstructor so we use typed pattern, example `_: User`
+
+      val label =
+        if (patternOnly.isEmpty) s"case $pattern =>"
+        else pattern
+      val cursorSuffix = (if (patternOnly.nonEmpty) "" else " ") +
+        (if (clientSupportsSnippets) "$0" else "")
       new TextEditMember(
         filterText = label,
```

Comment:
Same here, let's make sure that the completions are correctly shown in vs code

Reply:
From what I've tested, they work correctly. The code is mostly the same as for Scala 3 and there weren't any issues about it

Reply:
Ok, thanks for checking!